connectors/src/connectors/google_drive/temporal/activities.ts

```diff
@@ -675,24 +684,24 @@ export async function garbageCollector(
     files.map(async (file) => {
       return queue.add(async () => {
         try {
-          const isInFolder = await objectIsInFolder(
+          const isInFolder = await objectIsInFolders(
```

Comment:
If we're reunning on memoized information this migh not represent the truth and we may suppress stuff that we're supposed to sync (eg I sync A/ B/ but not C/ D/ and I move D/x under A/x) the memoized value for the parent of x (which is D) will cause x to be deleted, right?

My gut feeling is that we want memoization only during fullSync but not for incrementalSync or maybe reset it at the start of each workflow (assuming a long running incrementalSync) ?